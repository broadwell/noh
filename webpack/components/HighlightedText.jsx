import React from "react";
import PropTypes from "prop-types";
import TextLine from "./TextLine";

// containing component needs to know about currentTime and
// calculate currentPhraseID to pass into this component
class HighlightedText extends React.Component {
  createLines(linesType) {
    return this.props.phrases.map(phrase => (
      <TextLine
        key={phrase.phraseID}
        text={phrase[linesType]}
        active={phrase.phraseID === this.props.currentPhraseID}
      />
    ));
  }

  render() {
    const translationLines = this.createLines("translation");
    const transcriptionLines = this.createLines("transcription");
    return (
      <div className="transcription">
        <div className="transcription__title">
          <p>
            Singing style: <span>{this.props.singingStyle}</span>
          </p>
        </div>

        <div className="transcription__text">
          <div className="transcription__original">{transcriptionLines}</div>
          <div className="transcription__translation">{translationLines}</div>
        </div>
      </div>
    );
  }
}

HighlightedText.propTypes = {
  singingStyle: PropTypes.string.isRequired,
  currentPhraseID: PropTypes.string.isRequired,
  phrases: PropTypes.arrayOf(
    PropTypes.shape({
      phraseID: PropTypes.string,
      startTime: PropTypes.number,
      endTime: PropTypes.number,
      transcription: PropTypes.string,
      translation: PropTypes.string
    })
  ).isRequired
};

export default HighlightedText;
