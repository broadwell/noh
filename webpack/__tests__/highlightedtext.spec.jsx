import React from "react";
import { shallow } from "enzyme";
import HighlightedText from "../components/HighlightedText";

describe("<HighlightedText>", () => {
  it("renders as expected", () => {
    const phrases = [
      {
        phraseID: "I/1",
        startTime: 0,
        endTime: 200,
        transcription: "transcription line 1",
        translation: "translation line 1"
      },
      {
        phraseID: "I/2",
        startTime: 201,
        endTime: 410,
        transcription: "transcription line 2",
        translation: "translation line 2"
      }
    ];
    const component = shallow(
      <HighlightedText
        singingStyle="spoken"
        phrases={phrases}
        currentPhraseID="I/1"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
