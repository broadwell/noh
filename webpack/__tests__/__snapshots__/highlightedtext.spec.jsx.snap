// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HighlightedText> renders as expected 1`] = `
<div
  className="transcription"
>
  <div
    className="transcription__title"
  >
    <p>
      Singing style: 
      <span>
        spoken
      </span>
    </p>
  </div>
  <div
    className="transcription__text"
  >
    <Line
      active={true}
      key="I/1"
      transcription="transcription line 1"
      translation="translation line 1"
    />
    <Line
      active={false}
      key="I/2"
      transcription="transcription line 2"
      translation="translation line 2"
    />
  </div>
</div>
`;
